sherpa/astro/flux.py

```diff
@@ -529,12 +529,91 @@ def sample_flux(fit, data, src,
 
 def calc_sample_flux(id, lo, hi, session, fit, data, samples, modelcomponent,
                      confidence):
+    """Given a set of parameter samples, estimate the flux distribution.
+
+    This is similar to sample_flux but returns values for both the full
+    and component models (which can be the same). The set of parameter
+    samples is searched to remove rows where the parameters lie outside
+    the soft limits, and then this set of rows is used to calculate the
+    flux of the modelcomponent expression (the flux for the full model
+    is taken from the samples argument). The resulting set of fluxes is
+    used to calculate the median and the confidence range.
+
+    .. versionchanged:: 4.13.1
+       The clipping now includes parameters at the soft limits;
+       previously they were excluded which could cause problems with
+       parameters for which we can only calculate an upper limit.
+
+    Parameters
+    ----------
+    id : int or str
+        The dataset identifier. It must exist and have an associated model in
+        session.
+    lo : number or None, optional
+        The lower edge of the dataspace range for the flux calculation.
+        If None then the lower edge of the data grid is used.
+    hi : number or None, optional
+        The upper edge of the dataspace range for the flux calculation.
+        If None then the upper edge of the data grid is used.
+    session : sherpa.ui.utils.Session instance
```

Comment:
```suggestion
    session : sherpa.ui.utils.Session or sherpa.astro.ui.utils.Session instance
```

Reply:
Well, as sherpa.astro.ui.utils.Session is derived from sherpa.ui.utils.Session I don't feel the need to include all possible subclasses here! I have never felt comfortable with the existing documentation on how to document Python docstrings as I never feel I understand what to do for this very case (amongst other things)

Reply:
Accepted.